### PR TITLE
[AUD-1216] Update component structure

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

* Move all components in `components/general` into folders in `components/`
* Move page-components out from `containers` to `pages`
* Move containers into components that are not pages

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

This PR touches a *ton* of files

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

`npm run start:stage`

Went through common flows, sign up, upload, etc.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-aud-1216

File: src/components/artist/ArtistCard.js [hunk 1]
```diff
@@ -9,9 +9,9 @@ import {
 } from 'common/hooks/useImageSize'
 import { WidthSizes, SquareSizes } from 'common/models/ImageSizes'
 import DynamicImage from 'components/dynamic-image/DynamicImage'
-import FollowButton from 'components/general/FollowButton'
-import Stats from 'components/general/Stats'
-import UserBadges from 'containers/user-badges/UserBadges'
+import FollowButton from 'components/follow-button/FollowButton'
```
Comment (dylanjeffers) [inline src/components/artist/ArtistCard.js:12]: just a thought for the future, how would we feel about an import style like
```
import {FollowButton} from 'components/follow-button'
```
so essentially follow-button directory has an index that exports the api for this component/feature?

Comment (sliptype) [reply src/components/artist/ArtistCard.js:12]: I'm a fan of that, we're doing it in the mobile client

Comment (raymondjacobson) [reply src/components/artist/ArtistCard.js:12]: https://github.com/AudiusProject/protocol-dashboard/tree/main/src/components/AppBar we do it in the protocol dashboard as well

Comment (raymondjacobson) [reply src/components/artist/ArtistCard.js:12]: one thing at a time though 😓

**Review by sliptype — APPROVED** [review]
After looking at every file I can confidently say this looks good to me

**Review by dylanjeffers — APPROVED** [review]

**Review by Kyle-Shanks — APPROVED** [review]

**Pull request merged**